backend_new/src/services/app.service.ts

```diff
@@ -1,15 +1,106 @@
-import { Injectable } from '@nestjs/common';
+import fs from 'fs';
+import { join } from 'path';
+import {
+  Inject,
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+  OnModuleInit,
+} from '@nestjs/common';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { startCronJob } from '../utilities/cron-job-starter';
 import { SuccessDTO } from '../dtos/shared/success.dto';
 import { PrismaService } from './prisma.service';
 
+const CRON_JOB_NAME = 'TEMP_FILE_CLEAR_CRON_JOB';
+
 @Injectable()
-export class AppService {
-  constructor(private prisma: PrismaService) {}
+export class AppService implements OnModuleInit {
+  constructor(
+    private prisma: PrismaService,
+    @Inject(Logger)
+    private logger = new Logger(AppService.name),
+    private schedulerRegistry: SchedulerRegistry,
+  ) {}
+
+  onModuleInit() {
+    startCronJob(
+      this.prisma,
+      CRON_JOB_NAME,
+      process.env.TEMP_FILE_CLEAR_CRON_STRING,
+      this.clearCSV.bind(this),
+      this.logger,
+      this.schedulerRegistry,
+    );
+  }
 
   async healthCheck(): Promise<SuccessDTO> {
     await this.prisma.$queryRaw`SELECT 1`;
     return {
       success: true,
     } as SuccessDTO;
   }
+
+  /**
+    runs the job to remove existing csvs and zip files
+  */
+  async clearCSV(): Promise<SuccessDTO> {
+    this.logger.warn('listing csv clear job running');
+    await this.markCronJobAsStarted();
+    let filesDeletedCount = 0;
+    await fs.readdir(join(process.cwd(), 'src/temp/'), (err, files) => {
+      if (err) {
+        throw new InternalServerErrorException(err);
+      }
+      Promise.all(
+        files.map((f) => {
+          if (!f.includes('.git')) {
+            filesDeletedCount++;
+            fs.unlink(join(process.cwd(), 'src/temp/', f), (err) => {
+              if (err) {
+                throw new InternalServerErrorException(err);
```

Comment:
It does appear to cause a fatal crash on the system if this error happens. I tried wrapping it in a try catch and it still happens. The way I tested it is if you put a subdirectory into the temp folder with files in that and then run the cron job the error happens. I don't think that would happen ever, but somehow it happened at some point on my system. Most likely it was left over files from testing but 🤷 . Any thoughts?

Reply:
I changed some things up to better catch errors without completely breaking down and also change unlink to delete files and folders so we should be okay even if something wonky happens with the zipping process